watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
-                if all(k != i["name"] for i in self.water_analysis_input_list):
-                    self.water_analysis_input_list.append(v)
+        if flash_method == "vapor-fraction":
+            vapor_fraction_amount = (
+                {
+                    "unit": input_unit_set["vaporMolFrac"]["oli_unit"],
+                    "value": None,
+                },
+            )
+            if float(vapor_fraction):
+                vapor_fraction_amount.update({"value": float(vapor_fraction)})
+            else:
+                raise RuntimeError(
+                    f"Invalid vapor fraction: {vapor_fraction}. Expected number"
+                )
+            input_dict["vaporMolFrac"] = vapor_fraction_input
 
-        input_list = deepcopy(self.water_analysis_input_list)
-        return input_list
+        if flash_method == "isochoric":
+            volume_input = {
+                "unit": input_unit_set["totalVolume"]["oli_unit"],
+                "value": None,
+            }
+            if float(volume):
+                volume_input.update({"value": float(volume)})
+            else:
+                raise RuntimeError(f"Invalid volume: {volume}. Expected number")
+            input_dict["totalVolume"] = volume_input
+
+        if flash_method == "setph":
+            ph_input = {
+                "targetPH": {
+                    "unit": "",
+                    "value": None,
+                },
+            }
+            if float(ph):
+                ph_input["targetPH"].update({"value": float(ph)})
+            else:
+                raise RuntimeError(f"Invalid ph: {ph}. Expected number")
+            input_dict["targetPH"] = ph_input
+            if not acid_titrant:
+                acid_titrant = "HCl"
+            input_dict["pHAcidTitrant"] = get_oli_name(acid_titrant)
+            if not base_titrant:
+                base_titrant = "NaOH"
+            input_dict["pHBaseTitrant"] = get_oli_name(base_titrant)
+
+        if flash_method == "precipitation-point":
+            missing_args = [
+                arg for arg in [formed_solid, precipitant_inflow] if arg is None
+            ]
+            if missing_args:
+                raise RuntimeError(
+                    f"Missing argument(s) for {flash_method}: {missing_args}"
+                )
+            else:
+                input_dict.update(
+                    {
+                        "solidToPrecipitate": formed_solid,
+                        "inflowToAdjust": precipitant_inflow,
+                    }
+                )
 
-    def _oli_units(self, conc, unit):
-        """
-        Converts concentrations between specified units.
+        input_dict["inflows"] = inflows
 
-        :param conc: concentration value for a solute
-        :param unit: concentration unit for a solute
+        if flash_method == "corrosion-rates":
+            _logger.info(
+                f"Ensure DBS file uses 'AQ' thermodynamic framework to use Corrosion Analyzer"
```

Comment:
Can we raise exception instead here if thermodynamic framework is not AQ? I see you do that in `client` for calling corrosion-contact-surfaces function, but it is unclear whether that will _always_ precede this flash method call (I don't think it necessarily would for the unknowing user.)

Furthermore, we should have at least one test demonstrating a simple use-case that includes using `get_corrosion_contact_surfaces` and this `corrosion_rates` flash method to make sure we observe the expected behavior from this functionality.

Reply:
The logic could be added to run_flash since a DBS file is already required for that function